lib/flipper/group.rb

```diff
@@ -0,0 +1,19 @@
+module Flipper
+  class Group
+    attr_reader :name
+
+    def initialize(name, &block)
+      name = name.to_sym
+      if name.to_s.include?(Types::Group::SEPARATOR)
```

Comment:
Yup, it would certainly be surprising. And the clients will notice it right away, as soon as `Flipper.register` is called.